feat: withdraw custom amount from savings

rolznz opened the pull request:
Closes https://github.com/getAlby/hub/issues/575

This allows users to send explicit amounts, e.g. to do a boltz swap, but is kept in the advanced options as Alby Hub is primarily a lightning wallet.

In the withdraw page, an input and send all checkbox are added. Because of how transaction fee calculation in bitcoin works, send all vs sending an amount are two different actions. The user is also warned if they send all when they have channels open, as it will drain their anchor reserves.

![image](https://github.com/user-attachments/assets/101bddff-bbc1-462a-b181-ab8abdda1d8b)

![image](https://github.com/user-attachments/assets/e1451f32-a37c-4c79-bd6a-12ee11c8f8ff)

rolznz commented on frontend/src/screens/wallet/WithdrawOnchainFunds.tsx:
```diff
@@ -139,18 +152,19 @@ export default function WithdrawOnchainFunds() {
-          </Alert>
-        )}
+        {!!balances?.onchain.reserved &&
+          (sendAll || +amount > balances.onchain.total * 0.9) && (
```
use spendable instead of total

bumi commented:
btw. I find this screen a bit too complicated and I have never seen a wallet where I need to confirm my address and check a box.

why does it actually need to be that I am the owner of that address and having the seed phrase of it?

bumi commented on frontend/src/screens/wallet/WithdrawOnchainFunds.tsx:
```diff
@@ -159,6 +173,34 @@ export default function WithdrawOnchainFunds() {
+            <Input
+              id="amount"
+              type="number"
+              value={sendAll ? balances.onchain.total : amount}
```
are we sure "send all" will be that amount?

rolznz replied:
No, good catch!

rolznz commented:
> btw. I find this screen a bit too complicated and I have never seen a wallet where I need to confirm my address and check a box.

@bumi this was because one of our users sent their funds to the wrong address :cry: 

I'll update the copy a bit.

rolznz commented:
Updated UI:
![image](https://github.com/user-attachments/assets/32ae1a95-abed-43c1-b268-0651eddb6e4c)

![image](https://github.com/user-attachments/assets/77bf7b5e-9527-4c5e-bd27-f30ee2c6a157)

Anchor reserve variants:

![image](https://github.com/user-attachments/assets/b8442d12-a14b-4d05-8227-2bddb59ad04e)

![image](https://github.com/user-attachments/assets/21818b6c-8030-49d4-98a8-e94c3bcf323e)

Replaced checkbox and confirm input with a confirmation dialog:

![image](https://github.com/user-attachments/assets/0bc2e38e-edff-45c8-9366-6597216f3418)

![image](https://github.com/user-attachments/assets/980138dc-1e9a-4a7e-a551-fdbcd7f51734)

bumi commented:
that's a nice UI.
